docs/source/development/overview.md

```diff
@@ -0,0 +1,203 @@
+   for review.
+
+1. Check that the continuous integration testing passes and fix any issues causing test
+   failures.
+
+1. Request reviews from other package developers. A PR cannot be merged into `develop`
+   until at least one approving review has been added to the code. Reviews will often
+   suggest changes to the code and you should discuss those suggestions and implement
+   them.
+
+1. Once a PR has been approved, the PR can be merged into `develop` and the branch can
+   be deleted.
+
+## The package development environment
+
+The short descriptions below provide the key commands needed to set up your development
+environment and provide links to more detailed descriptions of code development for
+`pyrealm`.
+
+### Python environment
+
+You will need to install Python 3.10 or greater to develop `pyrealm`. We recommend using
+`pyenv` or `pyenv-win` to manage your Python installations but this is only really
+needed if you want to be able to test your code on different python versions.
+
+### Package management
+
+We use [`poetry`](https://python-poetry.org/docs/#installation) for dependency
+management and for managing development environments and you will need to install it.
+
+### Installing `pyrealm`
+
+To develop `pyrealm`, you will also need to install [`git`](https://git-scm.com/) and
+then clone the `pyrealm` GitHub repository.
+
+```sh
+git clone https://github.com/ImperialCollegeLondon/pyrealm.git
+```
+
+You can now use `poetry` to install the package dependencies. This is not just the
+package requirements for end users of the package, but also a wider set of tools used in
+package development. The `poetry` tools uses the
+[pyproject.toml](https://github.com/ImperialCollegeLondon/pyrealm/blob/develop/pyproject.toml)
+file to configure the dependencies that will be installed.
+
+```bash
+poetry install
+```
+
+Poetry uses a virtual environment for package development: all packages are installed to
+a stand-alone python environment that is only used for `pyrealm` development. This makes
+sure that the development environment is consistent across python versions and different
+developers. It does mean you need to put `poetry run` before command to run commands in
+that environment or `poetry shell` to start a new shell that specifically uses that
+environment.
+
+You should now be able to run the following command to see that `pyrealm` is installed
+and is showing the current version.
+
+```sh
+poetry run python -c "import pyrealm; print(pyrealm.__version__)"
+```
+
+### Installing and using `pre-commit`
+
+Development of the `pyrealm` package uses [`pre-commit`](https://pre-commit.com/). This
+is a python tool that runs a set of checks on `git` commits and stops the commit from
+completing when any of those checks fail. We use `pre-commit` to help catch a wide range
+of common issues and make sure that all code pushed to the GitHub repository meets some
```

Comment:
It would be nice to have an example of a common issue or maybe mention what it does NOT capture

Reply:
Hard to say what it doesn't capture, but I've added a brief description and moved the details link closer.